src/components/header/HeaderActions.vue

```diff
@@ -0,0 +1,30 @@
+<script setup lang="ts">
+import BaseButton from "@/components/base/BaseButton.vue";
+</script>
+
+<template>
+  <div class="actions">
+    <RouterLink to="#">
+      <BaseButton button-type="secondary">Register</BaseButton>
+    </RouterLink>
```

Comment:
So, this is a common pitfall in terms of accessibility 😄
We definitely *shouldn't* have buttons within the components - they have separate click handlers, and have separate focus - it might lead to many weird scenarios.

If you would like to have a button that looks like button but could act as a link...
We could do some funny shenanigans with the props. 👀 

We could let's say, pass a prop like the `to` and based on if we have it or not - use a router-link instead of a button in our template 💡
Same could be if we would want an anchor tag behavior for some external pages - detect if we have `href` and then use `a` tag.
Check out this section of the Vue docs for some inspiration https://vuejs.org/guide/essentials/component-basics.html#dynamic-components

Reply:
I will see what I can magic with your tips, not tried any of these approaches yet

Reply:
`<template>
  <template v-if="!props.to">
    <button :class="buttonClasses" @click="$emit('click', $event)">
      <slot />
    </button>
  </template>
  <template v-else>
    <RouterLink :to="props.to">
      <span :class="buttonClasses" @click="$emit('click', $event)">
        <slot />
      </span>
    </RouterLink>
  </template>
</template>
`
I think this is what you were trying to suggest to me?

Reply:
That's one way to do it, but the other is essentially to have something like following (while assuming that the `to` prop is defaulting to `null`).

```vue
<template>
  <component
    :is="componentTag"
    :to="to"
    :class="buttonClasses"
    @click="$emit('click', $event)"
  >
    <slot />
  </component>
</template>

<script>
// ...
const componentTag = computed(() => props.to !== null ? 'router-link' : 'button');
</script>
```